### PR TITLE
MAYA-107463 Use view-space normals and positions in UsdPreviewSurface to match

### PR DESCRIPTION
**Pull request by williamkrick**
the new behavior in USD.

The reflections are still not correct when using the view space normals and posisions, but this is at least a step in the right direction.

### CODE REVIEW

File: lib/mayaUsd/render/vp2ShaderFragments/shaderFragments.cpp [hunk 1]
```diff
@@ -106,6 +107,9 @@ static const TfTokenVector _FragmentNames = { _tokens->BasisCurvesCubicColorDoma
 
                                               _tokens->UsdPrimvarColor,
 
+                                              _tokens->UsdUVTexture,
```
Comment (williamkrick) [inline lib/mayaUsd/render/vp2ShaderFragments/shaderFragments.cpp:110]: I guess this was the merge conflict, I will fix is properly.

File: lib/mayaUsd/render/vp2ShaderFragments/NwToNv.xml [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+        </implementation>
+        <implementation render="OGSRenderer" language="HLSL" lang_version="11.0">
+            <function_name val="NwToNv" />
+            <vertex_source>
+                <![CDATA[
+float3 iNwToNv(
+    float4x4 ViewIT,
+    float3 Nw)
+{
+    return (ViewIT * (vec4(Nw, 0.0))).xyz;
```
Comment (JGamache-autodesk) [inline lib/mayaUsd/render/vp2ShaderFragments/NwToNv.xml:62]: For HLSL I would expect something that looks like:
```
return = mul(float4(Nw, 0.0), ViewIT).xyz;
```

File: lib/mayaUsd/render/vp2ShaderFragments/NwToNv.xml [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+        </implementation>
+        <implementation render="OGSRenderer" language="Cg" lang_version="2.1">
+            <function_name val="NwToNv" />
+            <vertex_source>
+                <![CDATA[
+float3 iNwToNv(
+    float4x4 ViewIT,
+    float3 Nw)
+{
+    return (ViewIT * (vec4(Nw, 0.0))).xyz;
```
Comment (JGamache-autodesk) [inline lib/mayaUsd/render/vp2ShaderFragments/NwToNv.xml:84]: And Cg is most probably:
```
return mul(ViewIT, float4(Nw, 0.0f)).xyz;
```

File: lib/mayaUsd/render/vp2ShaderFragments/usdPreviewSurfaceLightingAPI2.xml [hunk 4]
```diff
@@ -778,6 +778,8 @@ surfaceShader(vec3 Peye, vec3 Neye,
     result.outGlowColor = vec3(0.0, 0.0, 0.0);
     result.outMatteOpacity = vec3(opacity);
 
+    //result.outColor = Neye;
+
```
Comment (JGamache-autodesk) [inline lib/mayaUsd/render/vp2ShaderFragments/usdPreviewSurfaceLightingAPI2.xml:782]: Debugging output?

**Review by JGamache-autodesk — CHANGES_REQUESTED** [review]

**Review by JGamache-autodesk — APPROVED** [review]

**Pull request closed without merging**